firmware/src/adc.c

```diff
@@ -16,14 +19,19 @@ void adc_init() {
   // Set PF1 to analog mode
   gpio_pin_mode(GPIOF, 1, GPIO_MODE_ANALOG, 0, GPIO_PUPD_NONE, GPIO_OTYPE_PP);
 
+  // Disable the Deep Power Down mode
   ADC1->CR &= ~ADC_CR_DEEPPWD;
   msleep(10);
+  // Enable the internal ADC voltage regulator
   ADC1->CR |= ADC_CR_ADVREGEN;
   msleep(10);
 
-  // Disable ADC1
+  // Ensure ADC1 is disabled
   ADC1->CR &= ~ADC_CR_ADEN;
 
+  // Set ADC1 clock divider to 32 (5MHz) again
```

Comment:
I only did this twice because I wasn't sure of the correct place to do it. No harm in doing it twice though.